superset/common/query_context_processor.py

```diff
@@ -328,7 +378,25 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
                     "when using a Time Comparison."
                 )
             )
-        for offset in time_offsets:
+
+        columns = df.columns
+        time_grain = self.get_time_grain(query_context, query_object) or TimeGrain.DAY
```

Comment:
Why are we defaulting to a daily time grain? If it's mandatory, I'd rather return a 400 if it's missing.